packages/vite/src/shared/hmr.ts

```diff
@@ -199,6 +207,17 @@ export class HMRClient {
     })
   }
 
+  public addBuffer(message: string): void {
+    this.buffer.push(message)
+  }
+
+  public sendCustomMessages(): void {
+    if (this.connection.isReady()) {
+      this.buffer.forEach((msg) => this.connection.send(msg))
+      this.buffer = []
+    }
+  }
```

Comment:
Should this be named something like `queue(message)` and `flush()`? And maybe also expose a `send(message)` that does a `queue(message); flush()`

Reply:
From the naming side, they definitely need to have more words because HMR client is about handling module reloading as a whole, not just about communication. So I am fine with `queueMessage` (especially because we have queueUpdate`), but I am not sure what is the purpose of a separate flush method

Reply:
I mean renaming `sendCustomMessages()` to `flush()` (or `flushMessages()`). As in https://nodejs.org/api/http.html#requestflushheaders. Not having an extra method. The extra would be `sendMessages()` that does a `queueMessage()` then `flushMessages()`. For example, [here](https://github.com/vitejs/vite/pull/15486/files#diff-e29fe9494f319e31a0c387bcef6c87085df362ee9456419b83c54ca8621ea489R149) you have:
```js
    this.hmrClient.addBuffer(JSON.stringify({ type: 'custom', event, data }))
    this.hmrClient.sendCustomMessages()
```
that could be
```js
    this.hmrClient.sendMessage(JSON.stringify({ type: 'custom', event, data }))
```
if we have the three methods (`sendMessage`, `queueMessage`, `flushMessages`)

Reply:
Updated

Reply:
Updated again 😄